### PR TITLE
Switch to vitest

### PR DESCRIPTION
**Pull request by remcohaszing**
This replaces Jest as a test runner with Vitest. Jest has too many issues resolving modules and running TypeScript.

The tests were renamed to match Vitest’s default test pattern.

The ESLint configuration was adjusted a bit to match the new setup as well.

### CODE REVIEW

File: src/Transloadit.js [hunk 1]
```diff
@@ -216,7 +216,7 @@ class TransloaditClient {
         checkResult(result)
 
         if (isResumable && Object.keys(allStreamsMap).length > 0) {
-          await sendTusRequest({
+          await tus.sendTusRequest({
```
Comment (mifi) [inline src/Transloadit.js:219]: why is this needed?

File: test/integration/live-api.test.js [hunk 2]
```diff
@@ -104,8 +101,6 @@ const genericOptions = {
   waitForCompletion: true,
 }
 
-jest.setTimeout(100000)
```
Comment (mifi) [inline test/integration/live-api.test.js:107]: what's the timeout in vitest? i believe this timeout was added because jest's default timeout was too low. unless vitest default timeout is at least 100 sec we should probably set the timeout for vitest too

**Review by mifi — COMMENTED** [review]
lgtm as long as you tested locally (live-api) and see that the number of tests are the same as before, and coverage still works

File: test/integration/live-api.test.js (hunk 2 above)
Comment (remcohaszing) [reply test/integration/live-api.test.js:107]: Good catch. I added back some timeouts. The syntax for this is a bit different in Vitest.

File: src/Transloadit.js (hunk 1 above)
Comment (remcohaszing) [reply src/Transloadit.js:219]: Module mocking works differently. In the follow-up PR that converts the project to TypeScript, will enable [`verbatimModuleSyntax`](https://www.typescriptlang.org/tsconfig/#verbatimModuleSyntax) (for correctness). That means destructuring `require()` is no longer possible anyway.

**Review by mifi — APPROVED** [review]
